features/networking/metrics.feature

```diff
@@ -63,3 +63,58 @@ Feature: SDN/OVN metrics related networking scenarios
     And the output should contain:
       | kubeproxy_sync_proxy_rules_duration       |
       | kubeproxy_sync_proxy_rules_last_timestamp |
+  
+  # @author anusaxen@redhat.com
+  # @case_id OCP-37704
+  @admin
+  Scenario: Should be able to monitor various ovnkube-master and ovnkube-node metrics via prometheus
+    Given I switch to cluster admin pseudo user
+    And I use the "openshift-ovn-kubernetes" project
+    And evaluation of `endpoints('ovn-kubernetes-master').subsets.first.addresses.first.ip.to_s` is stored in the :ovn_master_metrics_ep_ip clipboard
+    And evaluation of `endpoints('ovn-kubernetes-master').subsets.first.ports.first.port.to_s` is stored in the :ovn_master_metrics_ep_port clipboard
+    And evaluation of `cb.ovn_master_metrics_ep_ip + ':' +cb.ovn_master_metrics_ep_port` is stored in the :ovn_master_metrics_ep clipboard
+    
+    And evaluation of `endpoints('ovn-kubernetes-node').subsets.first.addresses.first.ip.to_s` is stored in the :ovn_node_metrics_ep_ip clipboard
+    And evaluation of `endpoints('ovn-kubernetes-node').subsets.first.ports.first.port.to_s` is stored in the :ovn_node_metrics_ep_port clipboard
+    And evaluation of `cb.ovn_node_metrics_ep_ip + ':' +cb.ovn_node_metrics_ep_port` is stored in the :ovn_node_metrics_ep clipboard
+    
+    Given I use the "openshift-monitoring" project
+    And evaluation of `secret(service_account('prometheus-k8s').get_secret_names.find {|s| s.match('token')}).token` is stored in the :sa_token clipboard
+    
+    Given evaluation of `%Q{curl -k -H \"Authorization: Bearer <%= cb.sa_token %>\" https://<%= cb.ovn_master_metrics_ep %>/metrics}` is stored in the :curl_query_for_ovn_master clipboard
```

Comment:
@anuragthehatter can this be query via and external route instead?

Reply:
Hmm..I am using the suggested way by Junqi Zhao who tests [prometheus](https://coreos.slack.com/archives/CH76YSYSC/p1608110887320000). We have done same for Openshift SDN as well. I guess the esternal route would expect kubeadmin creds

Reply:
@pruan-rht WDYT

Reply:
@anuragthehatter ok, if that's the only way, then I'm ok with it.  Just trying to get a cleaner look.

Reply:
not sure why we do the `eval` here first save it in the clipboard.  If it is tricky can we leave a comment?

other places seem to do interpolation once?
```ruby
      | exec_command_arg | -c                   |
      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.mac_ip %>:8443/metrics |

```

Reply:
This is okay for now, but to be annoying and pedantic, this step is used **91** times already.  At some point we should make a Prometheus step that uses `sa_token` clipboard by default, bypasses `bash -c` to avoid extra quoting and does other useful things.

```shell
features/tierN/web/monitoring/alerts-browser.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-ruler.openshift-user-workload-monitoring.svc:9091/alerts |
features/tierN/web/admin-console/dashboards.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=namespace%3Acontainer_cpu_usage%3Asum%7Bnamespace%3D%22<%= project.name %>%22%7D |
features/tierN/web/admin-console/dashboards.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=pod%3Acontainer_cpu_usage%3Asum%7Bpod%3D%27<%= pod.name %>%27%2Cnamespace%3D%27<%= project.name %>%27%7D |
features/tierN/web/admin-console/admin-console-api.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.console_operator_pod_id %>:8443/metrics |
features/tierN/web/admin-console/admin-console-api.feature:      | curl | -kH | Authorization: Bearer <%= cb.token %> | <%= cb.server %>/apis/user.openshift.io/v1/users/~ |
features/tierN/scheduler/alert.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" 'https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22PodDisruptionBudgetAtLimit%22%7D'|
features/tierN/scheduler/alert.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" 'https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22PodDisruptionBudgetLimit%22%7D'|
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=count%28container_fs_usage_bytes%29%20by%20%28node%29 |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.cmo_endpoint %>/metrics |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.po_endpoint %>/metrics |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=prometheus_remote_storage_shards |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://alertmanager-main.openshift-monitoring.svc:9094/api/v2/alerts?filter={alertname="Watchdog"} |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22Watchdog%22%7D |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/rules |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22KubePodNotReady%22%2Cnamespace%3D%22ocp-28961-proj%22%7D |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-ruler.openshift-user-workload-monitoring.svc:9091/alerts |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=version |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=version |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/rules |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/rules |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" https://thanos-querier.openshift-monitoring.svc:9093/api/v1/rules?namespace=ocp-32623-proj |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://alertmanager-main.openshift-monitoring.svc:9094/api/v2/alerts?filter={alertname="Watchdog"} |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22Watchdog%22%7D |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" 'https://thanos-querier.openshift-monitoring.svc:9092/api/v1/query?query=version&namespace=<%= cb.proj_name %>' |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" 'https://thanos-querier.openshift-monitoring.svc:9092/api/v1/query?query=version' |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" 'https://thanos-querier.openshift-monitoring.svc:9092/api/v1/query?query=version&namespace=ns1' |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=kube_pod_spec_volumes_persistentvolumeclaims_info |
features/tierN/logging/elasticsearch.feature:      | bash | -c | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" -H "Content-type: application/json" https://<%= cb.service_ip %>:<%= cb.service_port %>/_prometheus/metrics |
features/tierN/cloudcredentialoperator/cloudcredential.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/label/__name__/values|
features/tierN/cloudcredentialoperator/cloudcredential.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.cco_prom_ip %>:<%= cb.cco_prom_port %>/metrics|
features/tierN/apiserver/kube-apiserver.feature:    curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://localhost:8443/metrics > /tmp/OCP-21246.metrics
features/tierN/admin/etcd.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=etcd_server_quota_backend_bytes |
features/tierN/admin/etcd.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=etcd_mvcc_db_total_size_in_use_in_bytes |
features/tierN/admin/etcd.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=etcd_debugging_mvcc_db_total_size_in_bytes |
features/upgrade/monitoring/monitoring-upgrade.feature:    # curl -k -H "Authorization: Bearer $token" 'https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=cluster_installer'
features/upgrade/monitoring/monitoring-upgrade.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=cluster_installer |
features/upgrade/monitoring/monitoring-upgrade.feature:    # curl -k -H "Authorization: Bearer $token" 'https://alertmanager-main.openshift-monitoring.svc:9094/api/v1/alerts'
features/upgrade/monitoring/monitoring-upgrade.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://alertmanager-main.openshift-monitoring.svc:9094/api/v1/alerts |
features/upgrade/monitoring/monitoring-upgrade.feature:    # curl -k -H "Authorization: Bearer $token" https://grafana.openshift-monitoring.svc:3000/api/health
features/upgrade/monitoring/monitoring-upgrade.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://grafana.openshift-monitoring.svc:3000/api/health |
features/networking/sriov.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.mac_ip %>:8443/metrics |
features/networking/sriov.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.mac_ip %>:8443/metrics |
features/networking/metrics.feature:    #Running curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.metrics_ep %>/metrics if version is > 4.5 as sdn mmetrics should be using https scheme
features/networking/metrics.feature:    Given evaluation of `%Q{curl -k -H \"Authorization: Bearer <%= cb.sa_token %>\" https://<%= cb.metrics_ep %>/metrics}` is stored in the :curl_query_ge_4_6 clipboard
features/networking/metrics.feature:    #Running curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.metrics_ep %>/metrics if version is > 4.5 as sdn metrics should be using https scheme
features/networking/metrics.feature:    Given evaluation of `%Q{curl -k -H \"Authorization: Bearer <%= cb.sa_token %>\" https://<%= cb.metrics_ep %>/metrics}` is stored in the :curl_query_ge_4_6 clipboard
features/tierN/web/admin-console/dashboards.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=namespace%3Acontainer_cpu_usage%3Asum%7Bnamespace%3D%22<%= project.name %>%22%7D |
features/tierN/web/admin-console/dashboards.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=pod%3Acontainer_cpu_usage%3Asum%7Bpod%3D%27<%= pod.name %>%27%2Cnamespace%3D%27<%= project.name %>%27%7D |
features/tierN/web/admin-console/admin-console-api.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.console_operator_pod_id %>:8443/metrics |
features/tierN/web/admin-console/admin-console-api.feature:      | curl | -kH | Authorization: Bearer <%= cb.token %> | <%= cb.server %>/apis/user.openshift.io/v1/users/~ |
features/tierN/web/monitoring/alerts-browser.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-ruler.openshift-user-workload-monitoring.svc:9091/alerts |
features/tierN/scheduler/alert.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" 'https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22PodDisruptionBudgetAtLimit%22%7D'|
features/tierN/scheduler/alert.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" 'https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22PodDisruptionBudgetLimit%22%7D'|
features/tierN/admin/etcd.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=etcd_server_quota_backend_bytes |
features/tierN/admin/etcd.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=etcd_mvcc_db_total_size_in_use_in_bytes |
features/tierN/admin/etcd.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=etcd_debugging_mvcc_db_total_size_in_bytes |
features/tierN/cloudcredentialoperator/cloudcredential.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/label/__name__/values|
features/tierN/cloudcredentialoperator/cloudcredential.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.cco_prom_ip %>:<%= cb.cco_prom_port %>/metrics|
features/tierN/apiserver/kube-apiserver.feature:    curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://localhost:8443/metrics > /tmp/OCP-21246.metrics
features/tierN/logging/elasticsearch.feature:      | bash | -c | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" -H "Content-type: application/json" https://<%= cb.service_ip %>:<%= cb.service_port %>/_prometheus/metrics |
features/machine/machine.feature:      | exec_command_arg | curl -v -s -k -H "Authorization: Bearer <%= cb.token %>" <url> |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=count%28container_fs_usage_bytes%29%20by%20%28node%29 |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.cmo_endpoint %>/metrics |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.po_endpoint %>/metrics |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=prometheus_remote_storage_shards |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://alertmanager-main.openshift-monitoring.svc:9094/api/v2/alerts?filter={alertname="Watchdog"} |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22Watchdog%22%7D |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/rules |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22KubePodNotReady%22%2Cnamespace%3D%22ocp-28961-proj%22%7D |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-ruler.openshift-user-workload-monitoring.svc:9091/alerts |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=version |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=version |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/rules |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/rules |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" https://thanos-querier.openshift-monitoring.svc:9093/api/v1/rules?namespace=ocp-32623-proj |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://alertmanager-main.openshift-monitoring.svc:9094/api/v2/alerts?filter={alertname="Watchdog"} |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=ALERTS%7Balertname%3D%22Watchdog%22%7D |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" 'https://thanos-querier.openshift-monitoring.svc:9092/api/v1/query?query=version&namespace=<%= cb.proj_name %>' |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" 'https://thanos-querier.openshift-monitoring.svc:9092/api/v1/query?query=version' |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= user.cached_tokens.first %>" 'https://thanos-querier.openshift-monitoring.svc:9092/api/v1/query?query=version&namespace=ns1' |
features/tierN/monitoring/ci.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=kube_pod_spec_volumes_persistentvolumeclaims_info |
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=sum\(openshift_samples_retry_imagestream_import_total\) |
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=sum\(openshift_samples_failed_imagestream_import_info\) |
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/label/__name__/values | 
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=openshift_build_created_timestamp_seconds |
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=openshift_build_duration_seconds |
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=apiserver_v1_image_imports_total |
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/label/__name__/values | 
features/build/metrics.feature:      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=openshift_samples_failed_imagestream_import_info |
```

Reply:
Right @rbbratta . At least for networking , its in the plan for me. For all the teams it may not be easy. There are changes coming in 4.8 wrt ovn-metrics as per epics so I will be tweaking some code then and incorporate some.

Reply:
yeah, it would be good to enhance it in a future PR.